### DISCUSSION TITLE
779

### DISCUSSION THREAD

**Discussion by giscus** [jojoldu/blog-comments#951] [Comments]
# 779

API의 캐시 중에는 모든 사용자 (로그인/비로그인 관계없이) 동일한 응답값을 줘야하는 것들이 있다. 로그인/비로그인에 관계없이 항상 일정한 데이터가 사용자 UI에 필요한 경우이다. 이를테면 인프런 서비스의 상단 헤더에 노출되는 강의 카테고리 (1 depth, 2depth, 3depth) 같은 경우. 이 캐시 데이터는 서비스의 모든 페이지에서 호출하고 있다보니 페이지 조회수 만큼 캐시 API를 호출하게 된다. Redis 캐시 -> 로컬 캐시 등으로 캐싱처리를 해도 결국 서버로 가는 조회 요쳥수가 개선되는 것은 아니고, 서비스 전체에서 가장 많이 호출되는 API 영역이 개선된 것이 아니다. SSR을 CDN으로 캐시 해도, SSG로 정적 페이지를 만들어도 헤더 영역은 사용자별 로그인 구분을 위해 CSR로 구..

https://jojoldu.tistory.com/779

**Comment by jiwooproity** [comment]
소중한 하나의 경험을 공유해 주셔서 감사드립니다.
항상 유익한 정보, 인사이트 얻어가요. 게시글 잘 보고 있습니다!

**Comment by mystyle2006** [comment]
이렇게 서버를 분리해서 처리하게 되었을 때 해당 카테고리 버전 관리를 별도로 해야할 것 같은데 어떤 방식으로 진행하고 계신지 궁금합니다!
